src/averageAges.js

```diff
@@ -36,8 +45,22 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = !!withChildren
+    ? people.filter(person =>
+      person.sex === 'f' && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === FEMALE);
+
+  const womenTotalAge = women.reduce((acc, woman) => {
+    return acc + (woman.died - woman.born);
+  }, 0);
+
+  womenTotalAge.sort().reverse();
+
+  const averafeWoomanAge = womenTotalAge / women.length + 1;
+
+  return averafeWoomanAge;
 }
```

Comment:
The function name 'calculateWomenAverageAge' is correct, but there are some issues inside the function. On line 59, you are trying to sort and reverse 'womenTotalAge', which is a number, not an array. Also, on line 61, you are adding 1 to the average age for no apparent reason.